common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -311,6 +318,37 @@ static void copyMemory(Object src, long srcOffset, Object dst, long dstOffset, l
         }
     }
 
+    static boolean equals(final byte[] bytes1, final byte[] bytes2) {
+        if (bytes1.length != bytes2.length) {
+            return false;
+        }
+        if (bytes1.length == 0) {
+            return true;
+        }
+        int remainingBytes = bytes1.length & 7;
+        for (int i = bytes1.length - 8; i >= remainingBytes; i -= 8) {
+            final long word1 = UNSAFE.getLong(bytes1, ARRAY_BASE_OFFSET + i);
+            final long word2 = UNSAFE.getLong(bytes2, ARRAY_BASE_OFFSET + i);
+            if (word1 != word2) {
+                return false;
+            }
+        }
+        if (remainingBytes >= 4) {
+            remainingBytes -= 4;
+            final int word1 = UNSAFE.getInt(bytes1, ARRAY_BASE_OFFSET + remainingBytes);
+            final int word2 = UNSAFE.getInt(bytes2, ARRAY_BASE_OFFSET + remainingBytes);
+            if (word1 != word2) {
+                return false;
+            }
+        }
+        if (remainingBytes >= 2) {
+            final char word1 = UNSAFE.getChar(bytes1, ARRAY_BASE_OFFSET);
+            final char word2 = UNSAFE.getChar(bytes2, ARRAY_BASE_OFFSET);
+            return word1 == word2 && (remainingBytes == 2 || bytes1[2] == bytes2[2]);
```

Comment:
Is it ok to access element `2` if `remainingBytes == 2`?  Seems like this would be out of bounds.